### ISSUE TITLE
Flexibility of the model

### ISSUE THREAD

**Issue by teamdisc** [galavebalaji/travelbooks#40]
In Nimble, we prefer to use [JSON:API](https://jsonapi.org) when possible. By having `DataResponse` not generic, it means that we will need to create a lot of `DataResponse` for multiple types of data (e.g. user feeds).

 I think either we could use a library for decoding JSON:API, or we could create our own `JSONAPIDecoder` based on native `JSONDecoder`.

This is my last issue here. I saw you've put a lot of efforts in commenting your code. Thank you for your hard work and the code again! ✌🏻

**Comment by galavebalaji** [comment]
Agreed. For now I did not separate all things into small pieces for this scope. 
But We could have achieved it as per your suggestions. 

Thanks for your time and it was a great learning.
Looking forward for the final feedback. :)
